File: Python/Math-Array-String/18.py
# 9. Palindrome Number

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:  # Negative numbers are not palindromes
            return False
        reversed_x = str(x)[::-1]
        return str(x) == reversed_x

    

sol = Solution()
output = sol.isPalindrome(-121)
print(output)

File: JavaScript/Math-Array-String/18.js
// 9. Palindrome Number

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    
    if (x < 0) return false; 

    let str = x.toString(); // Convert number to string
    let reversedStr = str.split('').reverse().join(''); // Reverse the string
    return str === reversedStr;

};

console.log(isPalindrome(121));